bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/EnhancedRendition.java

```diff
@@ -79,22 +75,13 @@ public Dimension getDimension() {
                     LOG.error("Cannot parse rendition dimensions from metadata", nfex);
                 }
             } else {
-                Matcher matcher = pattern.matcher(getName());
-                if (matcher.matches()) {
-                    // Try to infer dimensions from rendition name
-                    try {
-                        int width = Integer.parseInt(matcher.group(WIDTH_GROUP));
-                        int height = Integer.parseInt(matcher.group(HEIGHT_GROUP));
-                        dimension = new Dimension(width, height);
-                    } catch (NumberFormatException nfex) {
-                        LOG.error("Cannot parse rendition dimensions from name", nfex);
-                    }
-                } else if (rendition.getSize() < (AdaptiveImageServlet.DEFAULT_MAX_SIZE ^ 2)) {
+                if (StringUtils.startsWith(rendition.getMimeType(), "image/")
+                    && rendition.getSize() < Math.pow(AdaptiveImageServlet.DEFAULT_MAX_SIZE, 2)) {
                     // Try to load image to determine dimensions, if not too large
                     try {
                         dimension = Imaging.getImageSize(getStream(), getName());
```

Comment:
You should use try with resources, otherwise the stream won't be closed properly!